### PR TITLE
Add snowflake from iptproxy

### PR DESCRIPTION
**Pull request by stevenmcdonald**
* Copy snowflake client code from https://github.com/tladesignz/IPtProxy
* Copy over their build changes to use a temp dir, see https://github.com/tladesignz/IPtProxy/pull/38

I haven't tested this or anything, but I think we can just sort of charge forward on snowflake work in the `main` branch. I didn't bring over the proxy code (yet?) to make the binary smaller since if we're not using it (yet?), but it should be just copy/paste the functions and imports

Curious if it builds for you, but otherwise, just a sanity check I think. You're welcome to try it out with Envoy if you want, though

### CODE REVIEW

File: IEnvoyProxy/go.mod [hunk 1]
```diff
@@ -11,15 +11,12 @@ replace (
 )
 
 require (
+	git.torproject.org/pluggable-transports/snowflake.git/v2 v2.0.0-00010101000000-000000000000
 	github.com/lucas-clemente/quic-go v0.27.2 // indirect
-	github.com/onsi/ginkgo v1.16.5 // indirect
-	github.com/onsi/gomega v1.17.0 // indirect
-	github.com/pion/dtls/v2 v2.1.5 // indirect
-	github.com/pion/sctp v1.8.2 // indirect
-	github.com/pion/transport v0.13.0 // indirect
```
Comment (mnbogner) [inline IEnvoyProxy/go.mod:19]: what were these dependencies from?

File: snowflake.patch [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ 	shutdown        chan struct{}
++	ClientConnectedCallback    func()
+ }
+ 
+ // Checks whether an IP address is a remote address for the client
+@@ -551,6 +552,8 @@ func (sf *SnowflakeProxy) runSession(sid string) {
+ 	select {
+ 	case <-dataChan:
+ 		log.Println("Connection successful.")
++		go sf.ClientConnectedCallback()
```
Comment (mnbogner) [inline snowflake.patch:104]: are this and SnowflakeClientConnected above defined in existing code?

File: IEnvoyProxy/go.mod (hunk 1 above)
Comment (stevenmcdonald) [reply IEnvoyProxy/go.mod:19]: probably DNSTT? I hasn't run `go mod tidy` in a while

File: snowflake.patch (hunk 2 above)
Comment (stevenmcdonald) [reply snowflake.patch:104]: Could you not find them? I didn't study the changes extensively, I just copied it over and tested that it compiled... so I'm pretty sure they must be.

**Review by mnbogner — APPROVED** [review]

**Comment by vitriolix** [comment]
nothing jumps out at me (without testing) so 🚢 ?

**Review by vitriolix — APPROVED** [review]

**Pull request merged**